conans/client/cmake.py

```diff
@@ -71,6 +72,33 @@ def _generator(self):
 
         raise ConanException("Unknown cmake generator for these settings")
 
+    def _cmake_compiler_options(self, target_os, target_arch, target_comp):
+        host_platform = platform.system()
+        same_platform = (target_os == host_platform)
+
+        cmake_flags = []
+        cmake_flags.append("-DCMAKE_SYSTEM_NAME=%s" % target_os)
+
+        if target_comp == "clang":
+            cmake_flags.append("-DCMAKE_C_COMPILER=clang")
+            cmake_flags.append("-DCMAKE_CXX_COMPILER=clang++")
+        elif target_comp == "gcc":
+            if target_os == "Windows" and not same_platform:
+                if target_arch == "x86":
+                    cmake_flags.append("-DCMAKE_C_COMPILER=i686-w64-mingw32-gcc")
+                    cmake_flags.append("-DCMAKE_CXX_COMPILER=i686-w64-mingw32-g++")
+                else:
+                    cmake_flags.append("-DCMAKE_C_COMPILER=x86_64-w64-mingw32-gcc")
+                    cmake_flags.append("-DCMAKE_CXX_COMPILER=x86_64-w64-mingw32-g++")
+            else:
+                cmake_flags.append("-DCMAKE_C_COMPILER=gcc")
```

Comment:
What about using toolchain files (using the option `-DCMAKE_TOOLCHAIN_FILE`) then? I thing this could give more flexibility, especially when deciding to use clang, mingw or cross-compiling to other architectures. 

I understand your concerns, but in general these overwrites take effect only if I set os="Windows" but my host platform is Linux. `CMAKE_CXX_COMPILER` does not overwrites $CXX but affects only the current project (so far as I know, I haven't found any evidence that CXX is overwritten)